### ISSUE TITLE
Uniform aliasing linter

### ISSUE THREAD

**Issue by acamargo** [clj-kondo/clj-kondo#2195]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

**Is your feature request related to a problem? Please describe.**

On the path to achieving codebase canonicity, you're likely to encounter the issue of having multiple aliases for the same namespace. This often arises due to the maintenance of old projects by different teams and individuals over time.

The lack of consistency across all files and references can significantly reduce the readability and maintainability of the codebase.

While we can take advantage of the [duplicate-require linter](https://github.com/clj-kondo/clj-kondo/blob/master/doc/linters.md#duplicate-require) to prevent different aliases for the same namespace within a single file, it'd be helpful a solution that checks all the namespaces across the entire project codebase.

You can also benefit from the [consistent-alias linter](https://github.com/clj-kondo/clj-kondo/blob/master/doc/linters.md#consistent-alias) when striving for canonicity. It ensures that specific, well-known namespaces are consistently aliased with the same name. However, you'll need to provide a canonical list of alias recommendations.

**Describe the solution you'd like**

I'm proposing a linter similar to `duplicate-require` that scans all namespaces within a project to identify inconsistent aliases without needing a predefined list of aliases.

**Additional context**
I've cooked a PR for this feature https://github.com/clj-kondo/clj-kondo/pull/2194

**Comment by borkdude** [comment]
This feature will only work when you use clj-kondo from the command line and lint a directory, but that's probably still useful.

One thing I'm wondering about if we should make this an option in `:consistent-alias`. 

Perhaps it's possible to fold the loop in linters.clj into another one which already walks all the namespaces, might save some time, not sure.

**Comment by acamargo** [comment]
> This feature will only work when you use clj-kondo from the command line and lint a directory, but that's probably still useful.

I've noticed the result depends on the files passed to the `--lint` indeed...
Running via the command line would be useful as I would be able to use it on CI.
But, I'd love to have that on my editor via Clojure-LSP, isn't that possible?

> One thing I'm wondering about if we should make this an option in `:consistent-alias`.

Indeed, that was my first approach. https://github.com/clj-kondo/clj-kondo/pull/2194/commits/393d39ac4da32b27c51e47eefedcff4a138a07fe
But I ended up creating a different linter because they behave differently, albeit the purpose is the same.

> Perhaps it's possible to fold the loop in linters.clj into another one which already walks all the namespaces, might save some time, not sure.

Hmm, I'm not sure if I understood your suggestion. I'll get some numbers to figure out the performance penalty.
